include/bluetooth/bluetooth.h

```diff
@@ -772,9 +803,218 @@ int bt_le_ext_adv_get_info(const struct bt_le_ext_adv *adv,
+ *
+ * Enabling the periodic advertisement can be done independently of extended
+ * advertisement, but both periodic advertisement and extended advertisement
+ * shall be enabled before any data is sent.
+ * Enabling the periodic advertisement while it is already
+ * enabled updates the random address.
+ *
+ * @param adv		Advertising set object.
+ * @param enable	Whether to enable or disable periodic advertisement.
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_enable(struct bt_le_ext_adv *adv, bool enable);
+
+struct bt_le_per_adv_sync_synced_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+
+	/** Periodic advertisement interval (N * 1.25MS) */
+	uint16_t interval;
+
+	/** Periodic advertisement advertiser clock accuracy (ppm) */
+	uint16_t clock_accuracy;
+
+	/** Advertiser PHY */
+	uint8_t phy;
+};
+
+struct bt_le_per_adv_sync_term_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+};
+
+struct bt_le_per_adv_sync_recv_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+
+	/** The TX power of the advertisement. */
+	int8_t tx_power;
+
+	/** The RSSI of the advertisement excluding any CTE. */
+	int8_t rssi;
+
+	/** The Constant Tone Extension (CTE) of the advertisement */
+	uint8_t cte;
+};
+
+struct bt_le_per_adv_sync_cb {
+	/** @brief The periodic advertising has been successfully been synced.
+	 *
+	 *  This callback notifies the application that the periodic advertising
+	 *  set has been successfully been synced, and will now start to
+	 *  receive periodic advertisement reports.
+	 *
+	 *  @param adv	The periodic advertising sync object.
+	 *  @param info	Information about the sync event.
+	 */
+	void (*synced)(struct bt_le_per_adv_sync *sync,
+		       struct bt_le_per_adv_sync_synced_info *info);
+
+	/** @brief The periodic advertising sync has been terminated.
+	 *
+	 *  This callback notifies the application that the periodic advertising
+	 *  sync has been terminated, either by local request, remote request or
+	 *  because due to missing data (e.g. by being out of range or sync).
+	 *
+	 *  @param adv	The periodic advertising sync object.
+	 */
+	void (*terminated)(struct bt_le_per_adv_sync *sync,
+			   const struct bt_le_per_adv_sync_term_info *info);
+
+	/** @brief Periodic advertisement data received.
+	 *
+	 *  This callback notifies the application of an periodic advertisement
+	 *  report.
+	 *
+	 *  @param adv	The advertising set object.
+	 *  @param buf	Information about the periodic advertisement event.
+	 *  @param buf	Buffer containing the periodic advertisement data.
+	 */
+	void (*recv)(struct bt_le_per_adv_sync *sync,
+		     const struct bt_le_per_adv_sync_recv_info *info,
+		     struct net_buf_simple *buf);
+};
+
+/** Periodic advertisement sync options */
+enum {
+	/** Convenience value when no options are specified. */
+	BT_LE_PER_SYNC_OPT_NONE = 0,
```

Comment:
Yeah, something got messed up there (I'm guessing a search-and-replace than went wrong). 

One abbreviation I've considered using for periodic advertisement is PA rather than PER_ADV. Would short a lot of names in this PR, but I'm not sure whether it's too much.

Reply:
I think PER_ADV is better, let's keep it.